### PR TITLE
Fixes #36841 - Switch to Process.clock_gettime for ping

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Don't you also need to change `duration_ms` itself?

**Comment by ekohl** [comment]
For those following along, quoting `man clock_gettime`:
> CLOCK_MONOTONIC
> A  nonsettable  system-wide  clock  that  represents monotonic time since—as described by POSIX—"some unspecified point in the past".  On Linux, that point corresponds to the num‐ ber of seconds that the system has been running since it was booted.
>
> The  CLOCK_MONOTONIC  clock  is  not  affected  by discontinuous jumps in the system time (e.g., if the system administrator manually changes the clock), but is  affected  by  the incremental  adjustments performed by adjtime(3) and NTP.  This clock does not count time that the system is suspended.  All CLOCK_MONOTONIC variants guarantee that the  time  re‐ turned  by consecutive calls will not go backwards, but successive calls may—depending on the architecture—return identical (not-increased) time values.

File: app/services/ping.rb [hunk 1]
```diff
@@ -36,11 +36,11 @@ def statuses
     private
 
     def duration_ms(start)
-      ((Time.new - start) * 1000).round.to_s
+      ((Process.clock_gettime(Process::CLOCK_MONOTONIC) - start) * 1000).round.to_s
```
Comment (ekohl) [inline app/services/ping.rb]: This made me look up what it returns. Turns out there is another optional argument for the unit. So this should work:
```suggestion
      (Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond) - start).to_s
```

Now I'm wondering if it makes sense to introduce a helper for this.

Comment (ehelms) [reply app/services/ping.rb]: I debated the same, since you've mentioned it I will add one.

**Review by ekohl — APPROVED** [review]

**Comment by ehelms** [comment]
[test katello]

**Pull request merged**